diff --git a/compiler/third_party/npu_compiler/src/vpux_compiler/src/dialect/VPUIP/transforms/passes/legalize_repeating_func_calls.cpp b/compiler/third_party/npu_compiler/src/vpux_compiler/src/dialect/VPUIP/transforms/passes/legalize_repeating_func_calls.cpp
index f5035ed74..37cad36b5 100644
--- a/compiler/third_party/npu_compiler/src/vpux_compiler/src/dialect/VPUIP/transforms/passes/legalize_repeating_func_calls.cpp
+++ b/compiler/third_party/npu_compiler/src/vpux_compiler/src/dialect/VPUIP/transforms/passes/legalize_repeating_func_calls.cpp
@@ -249,7 +249,7 @@ private:
             const bool firstTimeSeenValue = memorizedOutputs.insert(*x).second;
             return !firstTimeSeenValue;
         };
-        for (size_t i = 0; !argsAcrossCallSites.empty(); OperandRange2D::advance(argsAcrossCallSites), ++i) {
+        for (; !argsAcrossCallSites.empty(); OperandRange2D::advance(argsAcrossCallSites)) {
             const auto ithOutputs = argsAcrossCallSites.get();
             auto it = llvm::find_if(ithOutputs, visitValue);
             // this is currently not supported
